google/cloud/storage/internal/object_requests.cc

```diff
@@ -159,6 +159,49 @@ std::ostream& operator<<(std::ostream& os, UpdateObjectRequest const& r) {
   return os << "}";
 }
 
+ComposeObjectRequest::ComposeObjectRequest(std::string bucket_name,
+      std::string destination_object_name,
+      std::vector<ComposeSourceObject> source_objects,
+      ObjectMetadata destination_object_metadata)
+      : GenericObjectRequest(std::move(bucket_name),
+        std::move(destination_object_name)),
+        destination_metadata_(std::move(destination_object_metadata)),
+        source_objects_(std::move(source_objects)) {
+  using internal::nl::json;
+  json compose_object_payload_json;
+  compose_object_payload_json["kind"] = "storage#composeRequest";
+  compose_object_payload_json["destination"] = json::parse(
```

Comment:
Ugh, in `.JsonPayloadForCompose()` we basically create a `nl::json` convert to a string, then parse that string here, then convert to a `nl::json` again.  Maybe we can return a `nl::json` directly from the `ObjectMetadata::JsonPayloadForCompose()` member function?